paywall/src/unlock.js/web3Proxy.ts

```diff
@@ -42,22 +110,104 @@ export default function web3Proxy(
     window.web3 &&
     window.web3.currentProvider &&
     (window.web3.currentProvider.sendAsync || window.web3.currentProvider.send)
+  let proxyAccount: null | string = null
+  let proxyNetwork: string | number
+  let accountIframeReady = false
+  let currentLocks: Locks
+  let canUseUserAccounts: boolean = !window.web3
+  let hasNativeWeb3Wallet = !!window.web3
+
+  const useUnlockAccount = () =>
+    !hasNativeWeb3Wallet && proxyAccount && canUseUserAccounts
+
+  // we need to listen for the account iframe's READY event, and request the current account and network
+  const accountHandlers: MessageHandlerTemplates<MessageTypes> = {
+    [PostMessages.READY]: postMessage => {
+      return () => {
+        postMessage('account', PostMessages.SEND_UPDATES, 'account')
+        postMessage('account', PostMessages.SEND_UPDATES, 'network')
+      }
+    },
+    [PostMessages.UPDATE_ACCOUNT]: () => {
+      return account => {
+        proxyAccount = account
+        accountIframeReady = true
+      }
+    },
+    [PostMessages.UPDATE_NETWORK]: () => {
+      return network => {
+        proxyNetwork = network
+      }
+    },
+    [PostMessages.INITIATED_TRANSACTION]: postMessage => {
+      return () => {
+        // prompt the data iframe to refresh transactions
+        postMessage('data', PostMessages.INITIATED_TRANSACTION, undefined)
+      }
+    },
+    [PostMessages.SHOW_ACCOUNTS_MODAL]: (
+      _postMessage,
+      _dataIframe,
+      _checkoutIframe,
+      accountIframe
+    ) => {
+      return () => {
+        if (canUseUserAccounts) {
+          showIframe(window, accountIframe)
+        }
+      }
+    },
+    [PostMessages.HIDE_ACCOUNT_MODAL]: (
+      _postMessage,
+      _dataIframe,
+      _checkoutIframe,
+      accountIframe
+    ) => {
+      return () => {
+        hideIframe(window, accountIframe)
+      }
+    },
+  }
 
-  // TODO: this will hold the logic for enabling the user account wallet
   const checkForUserAccountWallet = async (
-    _: IframeType,
+    accountIframe: IframeType,
     postMessage: PostMessageToIframe<MessageTypes>
   ) => {
     // we don't have web3
-    hasWeb3 = false
+    // wait for locks to be retrieved
+    await waitFor(() => currentLocks)
+    // wait for the account iframe, then respond
+    await waitFor(() => accountIframeReady)
+    // we will use the proxy account!
+    if (!canUseUserAccounts) {
```

Comment:
There are a few places in this file where a variable is inverted, even though we already have another variable that represents the inversion of that variable. Is there another way to handle this?